sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -1171,9 +1186,7 @@ def _predict(self, X, check_input=True):
         return self._label_binarizer.inverse_transform(y_pred)
 
     def _score(self, X, y):
-        """Private score method without input validation"""
-        # Input validation would remove feature names, so we disable it
-        return accuracy_score(y, self._predict(X, check_input=False))
+        return super()._score_with_function(X, y, score_function=accuracy_score)
```

Comment:
it does not apply to classifier, but I did a small refactor so that 

```python
        # Input validation would remove feature names, so we disable it
        return accuracy_score(y, self._predict(X, check_input=False))
```

is only written once :)